### PR TITLE
#3261 Fix issue where blueprints are showing as dirty when selected with no changes

### LINKED ISSUE #3261: Blueprints show as edited once you click on it
Context
- https://pixiebrix.slack.com/archives/D029R8902JF/p1650943306268389
- The FormEditor for blueprints options normalizes the uiSchema
- Blueprints immediately show as having unsaved changes once you click on them

### PR DESCRIPTION
**Pull request by BLoe**
### Fixes #3261 

We shouldn't set defaults directly in the form state because those changes are hard to tell apart from user changes. Instead, let's normalize when a change is made.

### CODE REVIEW

**Comment by BLoe** [comment]
Still need to fix the snapshots/tests and add some tests for `normalizeSchema()`

**Review by BALEHOK — APPROVED** [review]

**Comment by BALEHOK** [comment]
Don't let the code coverage decrease!)

File: src/components/formBuilder/edit/FormEditor.tsx [hunk 1]
```diff
@@ -176,36 +167,43 @@ const FormEditor: React.FC<FormEditorProps> = ({
...
   };
 
-  // There's always at least 1 item in uiOrder array, "*".
+  // The uiOrder field may not be initialized yet
+  const order = uiOrder ?? ["*"];
   const canMoveUp =
     Boolean(activeField) &&
-    (uiOrder?.length > 2
-      ? uiOrder[0] !== activeField
+    (order.length > 2
```
[the remark this answers is not in the document]
Comment (BLoe) [reply src/components/formBuilder/edit/FormEditor.tsx:198]: Sorry, not sure I follow what you mean by this?

File: src/components/formBuilder/formBuilderHelpers.ts [hunk 2]
```diff
@@ -321,48 +321,31 @@ export const produceSchemaOnUiTypeChange = (
...
-      } else if (!draft.uiSchema[UI_ORDER].includes("*")) {
-        draft.uiSchema[UI_ORDER].push("*");
-      }
-      /* eslint-enable security/detect-object-injection */
-
-      if (needToUpdateRequired) {
-        draft.schema.required = [];
-      }
-    });
+export const normalizeSchema = (schema: Schema | undefined) => {
```
[the remark this answers is not in the document]
Comment (BALEHOK) [reply src/components/formBuilder/formBuilderHelpers.ts:324]: `normalizeSchema` is called from a callback of the `produce` function. `normalizeSchema` itself then invokes `produce`. what we get is:
```
produce(() => { // to creat new state
  produce(() => {}); // to normalize schema
});
```

File: src/components/formBuilder/edit/FormEditor.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (BALEHOK) [reply src/components/formBuilder/edit/FormEditor.tsx:200]:
```
const canMoveUp =
    Boolean(activeField) &&
    uiOrder != null &&
    (uiOrder.length > 2
      ? uiOrder[0] !== activeField
      : propertyKeys[0] !== activeField);
```

[the remark this answers is not in the document]
Comment (BLoe) [reply src/components/formBuilder/edit/FormEditor.tsx:200]: Yeah, that works 👍

File: src/components/formBuilder/formBuilderHelpers.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (BLoe) [reply src/components/formBuilder/formBuilderHelpers.ts:324]: Ohh, `produce` is called twice, I see what you mean, sorry.

**Pull request merged**